check/configuration/custom-linters/reference.md

```diff
@@ -0,0 +1,527 @@
+`download`: *string*. The download url. You must provide either runtime + packages or download, not both. Using runtimes is preferred. [See Runtimes](../../advanced-setup/runtimes.md)
+
+`issue_url_format`: string, a format string that accepts issue codes for links to issues docs.
+
+### `run_from_root_target`
+
+`run_from_root_target`: *string*. Walk up to find this file to detect the run from directory. 
+Prefer `run_from` at the command level.
+
+### `hold_the_line`
+
+`hold_the_line`: *optional boolean*. Whether hold-the-line will be done for this linter or not.
+
+### `read_output_from`
+
+`read_output_from`,  Tell the parser where to expect output from for reading (stdout, stderr, 
+tmp file). [See Output Sources](output-types.md#output-sources)
+
+### `prepare_command`
+
+`prepare_command`, ex. `[tflint, --init]`
+
+[//]: # (`version`, **???** )
+
+### `known_good_version`
+
+`known_good_version`: The version `trunk init` inits with. If not present Trunk will query the 
+version from its runtime or use the default version from the download.
+
+### `version_command`
+
+`version_command`:  Version check commands
+
+### `enabled`
+
+`enabled`: *optional boolean*. Whether this linter is enabled.
+
+### `batch`
+
+`batch`: *optional boolean*. Combine multiple files into the same execution.
+
+### `run_when`
+
+`run_when`, Should this linter be run only in CI? (or locally, vscode, monitor)
+
+### `in_place`
+
+`in_place`: *optional boolean*. Applies to formatters - does the formatter rewrite the file as 
+opposed to reading its contents from STDIN and outputting formatted contents to STDOUT?
+
+### `run_timeout`
+
+`run_timeout`: *duration string*. Describes how long a linter can run before timing out. 
+[See timeouts](../readme.md#timeout)
+
+### `disabled`
+
+`disabled`: *optional boolean*: Whether linter is actively disabled (and will not be recommended)
+and will not run (overrides enabled).
+
+### `commands`
+`commands`: Commands exposed by this linter. [See Linter Command Definition](#linter-command-definition)
+
+### `deprecated`
+
+`deprecated`: *string*. Deprecated information (set when linter is deprecated)
+
+### `known_bad_versions`
+
+`known_bad_versions`: *string list*. Versions of a linter that are known to be broken 
+and should not be run with Trunk. We will fall back to a `known_good_version` 
+if init or upgrade chooses something in this set.
+
+### `plugin_url`
+
+`plugin_url`: *string*, a plugin url for reporting issues.
+
+### `query_compile_commands`
+
+`query_compile_commands`, *optional boolean*
+
+### `idempotent`
+
+`idempotent`: *optional boolean*. Indicates whether a linter is idempotent w.r.t. config + 
+source code inputs.  e.g. semgrep fetches rules from the internet, so it's not idempotent . If 
+set, causes cache_ttl to default to 24h
+
+### `cache_ttl`
+
+`cache_ttl`, duration string, How long cached results are kept before they expire
+
+### `target`
+
+`target`: optional string, what target does this run on
```

Comment:
```suggestion
`target`: optional string, what target does this run on. Defaults to `${file}`.
```

We should expand on other examples here.

Reply:
I added three code examples.